jujugui/static/gui/src/test/test_app.js

```diff
@@ -909,18 +911,37 @@ describe('App', function() {
         // Disconnect and reconnect the WebSocket.
         conn.transient_close();
         conn.open();
-        assert.equal(1, conn.messages.length);
+        assert.equal(1, conn.messages.length, 'no login messages sent');
         assertIsLogin(conn.last_message());
         done();
       });
     });
 
+    it('tries to re-login with macaroons on disconnections', function(done) {
+      sessionStorage.clear();
```

Comment:
The whole file uses that this way, so yes I agree with you but not in this branch.